### PR TITLE
Bystander effect

### PR DESCRIPTION
**Pull request by joaponte**
Added bystander immune killing, added uptake of virus by immune cells, fixed bugged with lambda chemotaxis saturation (wrong z level in the original) and created a radius to seed immune cells to avoid immune cells appearing immediately on top of the first infected cell.

### CODE REVIEW

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py [hunk 1]
```diff
@@ -425,14 +440,14 @@ def step(self, mcs):
...
-            constant = 50.0 / 10
+            concentration = field[cell.xCOM, cell.yCOM, 1]
+            constant = lamda_chemotaxis
             l = constant / (1.0 + concentration)
             if cell.dict['activated']:
                 cd.setLambda(l)
             else:
                 cd.setLambda(0)
-
+            cd.setLambda(l)
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: doesn't this invalidate the else statement? Where you perhaps using this for some testing and forgot to take it out?

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py [hunk 2]
```diff
@@ -450,42 +465,48 @@ def step(self, mcs):
...
+                    cell.dict['ck_consumption'] = max_ck_consume  # TODO: replace by hill
+                    cell.dict['tot_ck_upt'] = 0
+
+                    self.cellField[x_seed:x_seed + int(cell_diameter), y_seed:y_seed + int(cell_diameter), 1] = cell
+                    cd = self.chemotaxisPlugin.addChemotaxisData(cell, "cytokine")
+                    if cell.dict['activated']:
+                        cd.setLambda(lamda_chemotaxis)
+                    else:
+                        cd.setLambda(0.0)
+                    cd.setLambda(lamda_chemotaxis)
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: same

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py [hunk 3]
```diff
@@ -450,42 +465,48 @@ def step(self, mcs):
...
+                    cd.assignChemotactTowardsVectorTypes([self.MEDIUM])
+                    cell.targetVolume = cell_volume
+                    cell.lambdaVolume = cell_volume
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:511]: lambda_volume = target_volume?? This was like this on the previous version, so we should really check for it. Do we have lambda_volume as a parameter somewhere?

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py [hunk 4]
```diff
@@ -646,17 +661,15 @@ def step(self, mcs):
...
+            # Added virus uptake
+            self.virus_secretor.uptakeInsideCellTotalCount(cell,max_ck_consume / cell.volume, 0.1)
 
             cell.dict['tot_ck_upt'] -= up_res.tot_amount  # from POV of secretion uptake is negative
-            # print('tot_upt',cell.dict['tot_ck_upt'],'upt_now',up_res.tot_amount)
+            print('tot_upt',cell.dict['tot_ck_upt'],'upt_now',up_res.tot_amount)
             if cell.dict['tot_ck_upt'] >= EC50_ck_immune:
                 cell.dict['activated'] = True
-
+            cell.dict['activated'] = True
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: again a forgotten testing thing

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py [hunk 5]
```diff
@@ -646,17 +661,15 @@ def step(self, mcs):
             # print(EC50_ck_immune)
             up_res = self.ck_secretor.uptakeInsideCellTotalCount(cell,
                                                                  max_ck_consume / cell.volume, 0.1)
+            # Added virus uptake
+            self.virus_secretor.uptakeInsideCellTotalCount(cell,max_ck_consume / cell.volume, 0.1)
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:669]: we should try to find the uptake rate for viruses. I'm pretty sure the viral concentrations and ck concentrations in our simulation differ quite a bit in values. And their units are not the same, I believe

**Review by JulianoGianlupi — CHANGES_REQUESTED** [review]
Only change needed is the removal of declarations used for testing

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 1 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: Yes. This was there for testing purposes. It should be removed.

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 2 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: Same as above.

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 4 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py]: Yes. Needs to be changed

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 3 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:511]: So I defined lambda volume to be same at target volume based on Macieks paper on cancer, but that number can be changed as its arbitrary.

Comment (JulianoGianlupi) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:511]: Interesting. ok

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 5 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:669]: This merits some discussion.

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py (hunk 3 above)
Comment (joaponte) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py:511]: But I think we can make it a more user adaptable by defining it on the heading (with all the other CompuCell parameters)

**Review by JulianoGianlupi — APPROVED** [review]

**Comment by JulianoGianlupi** [comment]
Just as a note, with the cytokine secretion rates used, as soon as an immune cell activates and starts to secrete ck they dominate the behavior. I'm going to play around with those values

**Review by tjsego — APPROVED** [review]
Looks great. Nice work. I'm approving.

**Pull request merged**